File: calc.js
#!/usr/bin/env -S node

process.stdin.on('data', line => {
	const stack = [];

	for (const e of line.toString().trim().split(' ')) {
		if (/^-?[0-9]+$/.test(e)) {
			stack.push(Number(e));
			continue;
		}

		if (stack.length < 2) {
			console.log("Too Few Arguments");
			return;
		}
		
		const b = stack.pop();
		const a = stack.pop();

		switch (e) {
			case "+": stack.push(a + b); break;
			case "-": stack.push(a - b); break;
			case "*": stack.push(a * b); break;
			case "/": stack.push(a / b); break;
			default : console.log("Invalid Token"); return;
		}
	}

	if (stack.length == 1)
		console.log(stack[0]);
	else
		console.log("Invalid Input");
});

File: calc.py
#!/usr/bin/env python3

while True:
	stack = []

	try: s = input()
	except KeyboardInterrupt: exit()

	for e in s.split(" "):
		if e.isdigit() or e.startswith("-") and e[1:].isdigit():
			stack.append(int(e))
			continue

		if len(stack) < 2:
			print("Too Few Arguments")
			break

		b, a = stack.pop(), stack.pop()

		match e:
			case "+": stack.append(a + b)
			case "-": stack.append(a - b)
			case "*": stack.append(a * b)
			case "/": stack.append(a // b)
			case _:
				print("Invalid Token")
				break

	if len(stack) == 1:
		print(stack[0])
	else:
		print("Invalid Input")
	
